src/buildParameters.ts

```diff
@@ -1,19 +1,17 @@
-import { glob } from 'glob';
 import { Uri, window, workspace } from 'vscode';
-import { Utils } from 'vscode-uri';
+import { URI, Utils } from 'vscode-uri';
 
 import * as fs from 'fs';
 import * as fse from 'fs-extra';
-import path from 'path';
+import { Connection, ConnectionManager } from './connectionManager';
 
 export type EvaluationParameters = {
   operationArgs: string[] | undefined;
   outputPath: Uri | undefined;
   testPath: Uri | undefined;
 };
 
-// Should be working with normalized data
-export function buildParameters(uri: Uri): EvaluationParameters {
+export function buildParameters(uri: Uri, expression: string | undefined | null): EvaluationParameters {
```

Comment:
Don't use both `undefined` and `null`, just use `undefined`. If `null` means something significant here and distinct from `undefined`, you may want to revisit the signature of this function.

Reply:
removed all nulls